.gitignore

```diff
@@ -6,3 +6,10 @@
 # Ignore binaries built by the makefile to avoid accidentally committing them
 /images/kubefed/hyperfed
 /images/kubefed/controller-manager
+
+# OSX trash
+.DS_Store
+
+# Files generated by JetBrains IDEs, e.g. IntelliJ IDEA
+.idea/
+*.iml
```

Comment:
`.gitignore` is used for project related files maintenance in `kubefed`.

I finally agreed this rule after thinking it over. I saw some projects which use `.gitignore` file with hundreds of lines code with editor specific ignore. It is hard to read and maintain. You can refer to https://github.com/kubernetes-sigs/kubefed/pull/871.

Reply:
I see the point. 
Actually, I copied the entries from the [Kubernetes main repository](https://github.com/kubernetes/kubernetes/blob/master/.gitignore), and probably worth to be inline with it. But this discussion is out of this PR scope. 
I removed the entries.